src/poly-op.h

```diff
@@ -9,6 +9,11 @@ poly_binary_int_fn_ptr new_poly_p_equal_na_equal(enum vctrs_type type);
 typedef bool (*poly_unary_bool_fn_ptr)(const void* x, r_ssize i);
 poly_unary_bool_fn_ptr new_poly_p_is_missing(enum vctrs_type type);
 
+struct poly_df_data {
+  enum vctrs_type* col_types;
+  const void** col_ptrs;
```

Comment:
For consistency we'd now write this right?

```suggestion
  enum vctrs_type* v_col_type;
  const void** v_col_ptr;
```